app/controllers/surveyor/responses_controller.rb

```diff
@@ -95,7 +95,8 @@ def create
     @permission_to_edit = @protocol_role.nil? ? false : @protocol_role.can_edit? if @protocol
 
     if @response.save
-      SurveyNotification.system_satisfaction_survey(@response).deliver_now if @response.survey.access_code == 'system-satisfaction-survey' && Rails.application.routes.recognize_path(request.referrer)[:action] == 'review'
+      action = Rails.application.routes.recognize_path(request.referrer)[:action] rescue nil
+      SurveyNotification.system_satisfaction_survey(@response).deliver_now if @response.survey.access_code == 'system-satisfaction-survey' && action == 'review'
```

Comment:
Let's change this to just use the `in_review?` helper to minimize our usage of `#recognize_path`